modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4717,9 +4724,8 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
         //}
         //codegenProperty.required = true;
 
-        String parameterDataType = this.getParameterDataType(parameter, parameterSchema);
-        if (parameterDataType != null) {
-            codegenParameter.dataType = parameterDataType;
+        if (parameterModelName != null) {
+            codegenParameter.dataType = parameterModelName;
```

Comment:
Why should codegenParameters handle dataType and complexType differently than codegenProperty?

Reply:
The change above simply uses a different (better) name in the code block as `parameterDataType` originally aims to store the model name so no change in code logic if I remember correctly.